src/sizeMapping.js

```diff
@@ -1,61 +1,76 @@
-    return adUnit.sizes;
-  }
-  const width = getScreenWidth();
-  if (!width) {
-    // size not detected - get largest value set for desktop
-    const mapping = adUnit.sizeMapping.reduce((prev, curr) => {
-      return prev.minWidth < curr.minWidth ? curr : prev;
-    });
-    if (mapping.sizes && mapping.sizes.length) {
-      return mapping.sizes;
-    }
-    return adUnit.sizes;
-  }
-  let sizes = '';
-  const mapping = adUnit.sizeMapping.find(sizeMapping => {
-    return width >= sizeMapping.minWidth;
-  });
-  if (mapping && mapping.sizes && mapping.sizes.length) {
-    sizes = mapping.sizes;
-    utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${sizes}`);
-  } else {
-    utils.logMessage(`AdUnit : ${adUnit.code} not mapped to any sizes for device width. This request will be suppressed.`);
-  }
-  return sizes;
+/**
+ * @param {Array<string>} labels
+ */
+export function setLabels(labels) {
+  configuredLabels = labels;
 }
+config.getConfig('labels', config => setLabels(config.labels));
 
-function isSizeMappingValid(sizeMapping) {
-  if (utils.isArray(sizeMapping) && sizeMapping.length > 0) {
-    return true;
-  }
-  utils.logInfo('No size mapping defined');
-  return false;
+/**
+ *
+ * @param {Array<SizeConfig>} config
+ */
+export function setSizeConfig(config) {
+  sizeConfig = config;
 }
+config.getConfig('sizeConfig', config => setSizeConfig(config.sizeConfig));
 
-function getScreenWidth(win) {
-  var w = win || _win || window;
-  var d = w.document;
-
-  if (w.innerWidth) {
-    return w.innerWidth;
-  } else if (d.body.clientWidth) {
-    return d.body.clientWidth;
-  } else if (d.documentElement.clientWidth) {
-    return d.documentElement.clientWidth;
+/**
+ * This function will return all relevant sizes when resolving the labels passed in vs active labels (active labels
+ * being either those enabled manually or specified and enabled by matchMedia queries)
+ *
+ * @param {Array} labels The set of labels
+ * @param {Array<Array>} initialSizes starting set of sizes
+ * @returns {Array<Array>} filtered set of sizes after resolving passed in labels against active labels
+ */
+export function resolveSizesFromLabels(labels, initialSizes = []) {
+  if (!Array.isArray(labels)) {
+    return initialSizes;
   }
-  return 0;
-}
+  let hasMatch = false;
+  let filteredSizes = configuredLabels
+    .map(label => ({
+      sizesSupported: initialSizes, // configured labels support all sizes that are passed in by default
+      labels: [label]
+    }))
+    .concat(sizeConfig)
+    .reduce((sizes, config) => {
+      // does this sizeMapping config match passed-in labels?
+      if (Array.isArray(config.labels) && labels.some(label => config.labels.includes(label))) {
+        if (typeof config.mediaQuery !== 'string') {
+          hasMatch = true;
+        } else {
+          // if no mediaQuery criteria or if matchMedia matches, filter sizes with sizesSupported
```

Comment:
should this comment be at line 56 since it refers to that "no mediaQuery criteria"?